src/ReachSets/ContinuousPost/BFFPS19/reach.jl

```diff
@@ -0,0 +1,264 @@
+using ..Utils: LDS, CLCDS
+
+using LazySets: CacheMinkowskiSum,
+                 isdisjoint
+
+import LazySets.Approximations: overapproximate
+import ProgressMeter: update!
+
+"""
+    reach(problem, options)
+
+Interface to reachability algorithms for an LTI system.
+
+### Input
+
+- `problem`   -- initial value problem
+- `options`   -- additional options
+
+### Output
+
+A sequence of [`ReachSet`](@ref)s.
+
+### Notes
+
+A dictionary with available algorithms is available via
+`Reachability.available_algorithms`.
+
+The numeric type of the system's coefficients and the set of initial states
+is inferred from the first parameter of the system (resp. lazy set), ie.
+`NUM = first(typeof(problem.s).parameters)`.
+"""
+function reach_mixed(problem::Union{IVP{<:CLDS{NUM}, <:LazySet{NUM}},
+                              IVP{<:CLCDS{NUM}, <:LazySet{NUM}}},
+               options::TwoLayerOptions
+              )::Vector{<:ReachSet} where {NUM <: Real}
+    # optional matrix conversion
+    problem = matrix_conversion(problem, options)
+
+    # list containing the arguments passed to any reachability function
+    args = []
+
+    # coefficients matrix
+    A = problem.s.A
+    push!(args, A)
+
+    # determine analysis mode (sparse/dense) for lazy_expm mode
+    if A isa SparseMatrixExp
+        push!(args, Val(options[:assume_sparse]))
+    end
+
+    n = statedim(problem)
+    blocks = options[:blocks]
+    partition = convert_partition(options[:partition])
+    T = options[:T]
+    N = (T == Inf) ? nothing : ceil(Int, T / options[:δ])
+    guards_proj = options[:guards_proj]
```

Comment:
Since this variable is only used once later, can you directly access `options[:guards_proj]` there instead?